pkg/config/config.go

```diff
@@ -129,3 +165,38 @@ func (gc *Config) configureProxies(httpProxy, httpsProxy, noProxy string) {
 		os.Setenv(strings.ToUpper(noProxyEnvVar), noProxy)
 	}
 }
+
+func addSecretDataToConfig(config *Config, secret *v1.Secret) {
+	extractedValues := map[string]string{}
+	keys := []string{caCertsKey, httpProxyKey, httpsProxyKey, noProxyKey, skipTLSVerifyKey}
+	for _, key := range keys {
+		if value, valueExists := secret.Data[key]; valueExists {
+			extractedValues[key] = string(value)
+		}
+	}
+
+	addDataToConfig(config, extractedValues)
+}
+
+func addConfigMapDataToConfig(config *Config, configMap *v1.ConfigMap) {
+	addDataToConfig(config, configMap.Data)
+}
+
+func addDataToConfig(config *Config, data map[string]string) {
```

Comment:
this could be a private method on the Config struct. then no need to pass config as first arg.

same for related methods.

Reply:
Added